### PR TITLE
feat: migrating to tap 15.0.0

### PR DESCRIPTION
**Pull request by makuzaverite**
fix #2971 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Eomm** [comment]
I think this PR should target the `next` branch because tap 15 drops node10 support and fastify v3 doesn't

**Comment by jsumners** [comment]
> I think this PR should target the `next` branch because tap 15 drops node10 support and fastify v3 doesn't

No, 10 got added back at the last minute -- https://github.com/tapjs/node-tap/pull/728#issuecomment-808437496

File: .taprc [hunk 1]
```diff
@@ -1,4 +1,3 @@
-esm: false
```
Comment (jsumners) [inline .taprc:1]: This PR is going to require _way_ more than just this one deletion. The `beforeEach` and `afterEach` methods no longer support callbacks. Please see https://github.com/tapjs/node-tap/pull/728#issue-601157222 for all breaking changes.

**Comment by dnlup** [comment]
@makuzaverite I think this got automatically closed because there's a new update of `tap`:

* https://github.com/fastify/fastify/pull/2974

**Pull request closed without merging**